base/shell/explorer/startmnucust.cpp

```diff
@@ -66,40 +66,213 @@ VOID OnClearRecentItems()
         }
         while (FindNextFileW(hPath, &info));
         FindClose(hPath);
-        /* FIXME: Disable the button*/
+        
+        EnableWindow(GetDlgItem(hwnd, IDC_CLASSICSTART_CLEAR), FALSE);
     }
 }
 
-INT_PTR CALLBACK CustomizeClassicProc(HWND hwnd, UINT Message, WPARAM wParam, LPARAM lParam)
+INT_PTR CALLBACK CustomizeModernProc(HWND hwnd, UINT Message, WPARAM wParam, LPARAM lParam)
 {
     switch(Message)
     {
         case WM_INITDIALOG:
-            /* FIXME: Properly initialize the dialog (check whether 'clear' button must be disabled, for example) */
         return TRUE;
+        case WM_COMMAND:
+            switch(LOWORD(wParam))
+            {
+                case IDOK:
+                    EndDialog(hwnd, IDOK);
+                    break;
+                case IDCANCEL:
+                    EndDialog(hwnd, IDCANCEL);
+                    break;
+            }
+            break;
+        default:
+            return FALSE;
+    }
+    return TRUE;
+}
+
+#define MAX_CLASSIC_OPTION 12
+
+BOOL ClassicOptDirty[] =
```

Comment:
(Comment for this list and the one below.)
You may instead use a structure that contains both information like this:
```c
static struct
{
    PCWSTR OptionName;
    BOOL IsDirty;
}
ClassicOpt[] =
{
    {L"NameOfOption1", FALSE},
    // etc...
};
```
especially if this is some kind of "global state" inside explorer.exe (and not something that should be per-instance of a class).